amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamARNUtilTest.java

```diff
@@ -0,0 +1,70 @@
+package software.amazon.kinesis.common;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.regions.Region;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ StreamARNUtil.class })
+public class StreamARNUtilTest {
+    private static final String STS_RESPONSE_ARN_FORMAT = "arn:aws:sts::%s:assumed-role/Admin/alias";
+    private static final String KINESIS_STREAM_ARN_FORMAT = "arn:aws:kinesis:us-east-1:%s:stream/%s";
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        PowerMockito.spy(StreamARNUtil.class);
+    }
+
+    @Test
+    public void testGetStreamARNHappyCase() throws Exception {
+        String streamName = "testStreamName";
+        String accountId = "123456789012";
+        PowerMockito.doReturn(Optional.of(Arn.fromString(String.format(STS_RESPONSE_ARN_FORMAT, accountId))))
```

Comment:
Unfortunately, this avoids testing the `getStsCallerArn()` method and, instead, assumes the method is correct. An error in the method will go unnoticed until run-time (or exposed by another test). Considering that said method is a part of `StreamARNUtil`, the class directly under test, these tests are the place to prove said method works.